nautilus_trader/backtest/data_loader.py

```diff
@@ -711,21 +728,55 @@ def _query(
 
         path = f"{self.root}/{filename}.parquet/"
         if not self.fs.exists(path):
-            return
+            if raise_on_empty:
+                raise FileNotFoundError
+            else:
+                return pd.DataFrame()
 
         dataset = ds.dataset(path, partitioning="hive", filesystem=self.fs)
-        df = dataset.to_table(filter=combine_filters(*filters)).to_pandas().drop_duplicates()
+        table = dataset.to_table(filter=combine_filters(*filters))
+        df = table.to_pandas().drop_duplicates()
+        mappings = self._read_mappings(path=path)
+        for col in mappings:
+            df.loc[:, col] = df[col].map({v: k for k, v in mappings[col].items()})
```

Comment:
Why is this being remapped?

Reply:
We (posssibly) transform the partition key to remove any values that may mess with filesystems (`/`, `.` etc.) as the partition key is written into the directory structure.

When we reload the data, we need to "undo" this via this mapping

Reply:
Oh okay, thanks for the context. Is the idea with the data loader to load data >> save it to disk as another format >> load the new format/share it?

Reply:
Yes but also to improve performance when loading the data into backtests. We can store in parquet and have much faster loading times than JSON or CSV (as well as better filtering abilities).

The idea is that we will eventually write a parquet -> nautilus loader in cython which will be much faster that parsing text.

Reply:
Awesome, that's what I thought, just wanted to make sure.